contribute/design/1.x/cstor/zcmd-operation.md

```diff
@@ -0,0 +1,232 @@
+---
+oep-number: ZCommands Execution Via CSP Custom Resource
+title: ZCommands execution via custom resources
+authors:
+  - "@mittachaitu"
+owners:
+  - "@kmova"
+editor: "@mittachaitu"
+creation-date: 2020-08-26
+last-updated: 2020-08-26
+status: provisional
+---
+
+# ZCommands execution via Custom Resource
+
+## Table of Contents
+
+- [ZCommands execution via Custom Resource](#zcommands-execution-via-custom-resource)
+	- [Table of Contents](#table-of-contents)
+	- [Summary](#summary)
+	- [Motivation](#motivation)
+		- [Goals](#goals)
+	- [Proposal](#proposal)
+		- [Stories](#stories)
+		- [Proposed Implementation](#proposed-implementation)
+		- [Steps to be performed by high level operator](#steps-to-be-performed-by-high-level-operator)
+		- [Low Level Design](#low-level-design)
+			- [WrokFlow](#wrokflow)
+
+## Summary
+
+This proposal brings out the design details to implement zpool/zfs commands execution
+via CSP custom resource.
+
+## Motivation
+
+- Zpool and ZFS command execution via CSP custom resource
```

Comment:
If we can have an elaboration of why this is required, What are the motivations? I can see that SPC --> CSPC migration kind of needs this. If we can put the exact story of that -- it would be a good example.

Reply:
Done